cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py

```diff
@@ -0,0 +1,152 @@
+""" API Views for course advanced settings """
+
+import edx_api_doc_tools as apidocs
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+from cms.djangoapps.models.settings.course_metadata import CourseMetadata
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists, view_auth_classes
+from ..serializers import CourseAdvancedSettingsSerializer
+from ....views.course import update_course_advanced_settings
+
+
+@view_auth_classes(is_authenticated=True)
```

Comment:
[This decorator](https://github.com/edx/edx-platform/blob/09d58a4eb84e38286a31ed82fb0f3699331de0b6/openedx/core/lib/api/view_utils.py#L112-L132) does apply authentication classes automatically. Since the same classes need to be specified again and again, this just encapsulates it in a common decorator function.